deployer/src/persistence.rs

```diff
@@ -45,13 +45,18 @@ impl Persistence {
                 state INTEGER          -- Enum indicating the current state of the deployment.
             );
 
+            CREATE TABLE IF NOT EXISTS libs (
+                name TEXT PRIMARY KEY, -- Name of the service.
+                lib TEXT               -- The name of linked library produced the last time the service was built.
+            );
```

Comment:
Maybe we can remove this table if we give each incoming deployment an id and store the id in the deployments table? Then we can use the same id when renaming the 'so' file?

Reply:
@chesedo Sorry if I'm misunderstanding, but are you saying that each deployment would then just have a single '.so' filename even after re-deployment? I don't think that would work due to our library caching/reloading issue (i.e., each new compiled '.so' file needs to have a unique name to be correctly loaded) :/

Reply:
Yes, each deployment would have a single 'so' for that deployment. This of course assumes we are sticking with the terminology of a deployment being a version of a service

Reply:
So technically I guess we should have 3 tables:
- Service (S) with the service name
- Deployments (D) with the deployment id. Here exists a (D) many-to-one (S) link
- Logs (L) with the logs for a deployment. Here exists a (L) many-to-one (D) link

The last two tables can always store the service name again if we want the DB view to remain easy (I guess the S table is then no longer needed really)